src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs

```diff
@@ -42,7 +42,9 @@ public async Task AnalyzeAsync(IProject project, IDependencyAnalysisState state,
                 throw new ArgumentNullException(nameof(state));
             }
 
-            foreach (var packageReference in state.Packages)
+            var packages = state.Packages.ToList();
```

Comment:
I wonder if we were to back the packages with an immutable collection if it would work without copying. For now, this is probably the easiest fix. Can you add a comment why we're doing a `.ToList()` here?

Reply:
Added